stock_secondary_unit/views/stock_move_views.xml

```diff
@@ -12,7 +12,9 @@
                 <field name="secondary_uom_qty" />
                 <field
                     name="secondary_uom_id"
-                    domain="[('product_tmpl_id.product_variant_ids', 'in', [product_id])]"
+                    domain="['|', ('product_id', '=', product_id),
+                                '&amp;', ('product_tmpl_id.product_variant_ids', 'in', [product_id]),
+                                         ('product_id', '=', False)]"
```

Comment:
```suggestion
                    domain="['|', ('product_id', '=', product_id),
                                ('product_tmpl_id.product_variant_ids', 'in', [product_id])]"
```



If the first part of the domain is evaluated as True, I think the second part of the domain is not checked because True or whatever is equal to True, so in case the partner_id is filled, it would not check this second part of the domain and it would only check in case product_id = False.

Reply:
It's correct this way, as `product_id` is always going to have a value, so if the product variant is filled in the secondary unit, it has to match. If not, the double condition of the second part is avoiding to match on other secondary units having a specific variant.